proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -7,112 +7,236 @@
+If WiFi access point credentials are automatically generated by HMI, the password could contain special characters or it could be just too long. It would be inconvenient for the user to enter such passwords in HMI. Another possible scenario is, HMI could keep changing passwords to improve the security of the network. In such cases, a user might need to connect to WiFi network every time password is changed.
 
 ## Proposed solution
 
 The available transport type for Secondary Transport can be configured through smartDeviceLink.ini file. 
 
-If `MultipleTransportsEnabled` is set as `true` and `SecondaryTransportForBluetooth` is `WiFi` in `smartDeviceLink.ini` configuration file. HMI should create a WiFi Access Point and the WiFi credentials should be communicated to the mobile proxy.
+If `MultipleTransportsEnabled` is set as `true` and `Secondary Transport` is `WiFi` in `smartDeviceLink.ini` configuration file. HMI could create a WiFi Access Point and the WiFi credentials can be communicated to the mobile proxy.
 
 Android and iOS provide APIs to connect to a WiFi network in case WiFi SSID and password is known. These APIs could be used in Mobile Proxy to connect to the WiFi access point created by HMI. The flow for the proposed solution would be:
 
-`HMI creates WiFi Access Point --> HMI provides WiFi Access point credentials to SDL Core --> SDL Core shares WiFi credentials using Primary Transport --> Mobile Proxy receives WiFi access point credentials and check if it is already connected --> If it's not connected, Mobile Proxy automatically connects to the WiFi access point.`
+`HMI creates WiFi Access Point --> HMI provides WiFi Access point credentials to SDL Core --> SDL Core shares WiFi credentials using Primary Transport --> Mobile Proxy receives WiFi access point credentials  --> Mobile Proxy automatically connects to the WiFi access point.`
 
-### Changes for communicating Credentials from HMI to Core
+### HMI API changes
 
-HMI will create a WiFi access point and communicate these using below HMI API.
+1. HMI will create a WiFi access point and communicate these using below HMI API.
+2. HMI will notify SDL Core using `OnWiFiTransportStatusUpdate` whenever WiFi state is changed.
+3. Device Info is changed to provide WiFi auto-connect feature support information to HMI. 
+4. HMI will receive `DeviceInfo.supportWiFiAutoConnect` in `RegisterAppInterface` Request using `primary transport`.
+5. HMI can send `OnSystemRequest` notification with `RequestType` as `CONNECT_WIFI` to inform app to try connecting WiFi. This notification will be sent only when a user has provided consent to share WiFi credentials, and the connected app supports WiFi auto-connect and the user attempts to launch that app.
+6. HMI will not send `OnSystemRequest` notification with `RequestType` as `CONNECT_WIFI` if app is already connected using WiFi.
 
 ```

```

Comment:
Is it possible to add the language you're using here, so that the code block is easier to read?  For example: ```xml

Here's how the code block could look: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0244-setmediaclocktimer-custom-playback-rates.md#mobile_api

Reply:
Added changes in [this ](https://github.com/smartdevicelink/sdl_evolution/pull/819/commits/8cecbd20a37e7db3d90e66bb3a338fabd1aa3c93)commit.